src/Controllers/orderController.ts

```diff
@@ -0,0 +1,40 @@
+import { createOrder, processOrderItem, getUserIdFromSession} from '../services/orderService';
+import { placeOrderSchema } from '../validators/ordersSchema';
+
+const db = require('../Database/Models/index.ts');
+
+export const placeOrder = async (req, res) => {
+  const transaction1 = await db.sequelize.transaction();
+
+  try {
+    await placeOrderSchema.validateAsync(req.body);
+    const sessionId = req.headers['authorization'] as string;
+    const userID = await getUserIdFromSession(sessionId);
+    const products = req.body;
+
+    const newOrder = await createOrder(userID, transaction1);
+
+    for (const item of products) {
+      await processOrderItem(item, newOrder, transaction1);
+
+    }
+    await transaction1.commit();
+
+    res.status(200).json(newOrder);
+    } catch (error: any) {
+      await transaction1.rollback();
+      if(error.isJoi === true){
+        error.status = 422;
```

Comment:
why 422 ? why not regular 400 ?
422 often used when the client sends a data with invalid format, like wrong json format